### DISCUSSION TITLE
프로퍼티관리

### DISCUSSION THREAD

**Discussion by Dae-Hwa** [Sc0up/scoup-backend#87] [Refactor]
@janeljs @min27604 
참고자료 - https://www.baeldung.com/properties-with-spring

현재 프로퍼티가 사용하는 클래스 위에 붙어있는데요,
PropertySource 어노테이션은 Configuration 위에 놓는게 더 적절하다고 하는 것 같아요
스프링 공식 문서를 봐야 더 정확하겠지만, application.properties에 설정하는 것과 비슷한 느낌인 것 같은데,
개선을 해야되는 부분이 맞는건지 검토해봐야할 것 같습니다.

의견 부탁드립니다!
추가적인 자료나 읽을거리 같은거 공유해주셔도 좋아요~

**Comment by janeljs** [comment]
저도 property source 어노테이션 처음 사용했을 때 찾아보는데 보통 configuration이랑 같이 쓰기는 하더라구요! 그래서 클래스 위에 둬도 잘 작동하나 싶었는데 일단 잘 작동해서 그대로 뒀던 거 같아요

**Comment by janeljs** [comment]
`spring.config.import=classpath:auth.properties`와 같이 관리하는 방식으로 변경하였습니다.

**Comment by Dae-Hwa** [comment]
ConfigurationProperties 

https://www.baeldung.com/properties-with-spring#5-hierarchical-properties
https://www.baeldung.com/configuration-properties-in-spring-boot

https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config
https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config.typesafe-configuration-properties.vs-value-annotation

**Reply by janeljs** [reply]
현재 GitHubProperty, GoogleProperty, KakaoProperty가 OAuthProperty를 상속하는 구조로 구현이 되어있어서 상속받는 필드에 대해 `@ConfigurationProperties`  적용이 안되는데 방법이 없을지 고민해볼게요ㅎㅎ

**Reply by janeljs** [reply]
https://stackoverflow.com/questions/50959094/spring-boot-extend-existing-properties-class-with-new-prefix/50959867#50959867

**Reply by janeljs** [reply]
https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config.typesafe-configuration-properties.constructor-binding
